### DISCUSSION TITLE
Python Agent：have service instance but don't have trace data

### DISCUSSION THREAD

**Discussion by Mushano** [apache/skywalking#10810] [Q&A] [answered]
### Python Agent version
1.0.1

### Python version
3.7

### Description of the problem
I deploy the django service using non-intrusive deployment, and I can find that there are already corresponding service names and service instances in the UI of skywalking, but there will be no trace data when the interface is called.
![image](https://github.com/apache/skywalking/assets/38551745/2b3b86b8-c6ac-4c0d-bcde-e6bdb0126abf)
![image](https://github.com/apache/skywalking/assets/38551745/fcb96a83-3146-400c-b7b3-c76087de966a)
（You can see that only the python agent doesn't work,java agent work well）

### Purpose
What I want to know is, if Python agents can connect to OAP, why can't trace data be tracked? Is this a bug?

**Comment by Superskyyy** [comment, marked as answer]
@Mushano Hi, which agent version and django version are you using? All plugins including Django is tested in CI. We also have extensive e2e testing to ensure the agent core functionalities are working. 

Please first turn on the debug log of agent (see documentation agent_logging_level -> 'DEBUG') and check if a segment can be reported upon access, you should see something like "reporting segment blabla whenever you access the django app endpoint". https://skywalking.apache.org/docs/skywalking-python/next/en/setup/configuration/ 

Another possibility is that maybe double check the time range setting in the skywalking UI, that can sometimes go wrong.

**Reply by Mushano** [reply]
I really appreciate your answer!!
 I use the following version information: 
- Skywalking Python Agent: 1.0.1
- skywalking:9.3
- django：2.1.4

I follow the step your adviced that turn on the debug log, then I try to call the interface about the django service.

Database:h2
It seems that nothings be reported upon access.
![image](https://github.com/apache/skywalking/assets/38551745/1d184644-64cf-4005-8d1f-ebda24a41b8f)

Database:elasticsearch
Although I can see that some segment data is reported when I reboot the service,nothing trace data in the ui.
like this:
```
2023-05-16 15:03:45,977 skywalking [pid:42] [Thread-4] [DEBUG] reporting segment Segment@140514892420368[segment_id=8359a0cef3b711eda05b161bd37a2426, spans=[<skywalking.trace.span.EntrySpan object at 0x7fcc2c3f99d0>], timestamp=1684220502660, related_traces=[<skywalking.trace.segment._NewID object at 0x7fcc2c3f9690>]]
2023-05-16 15:03:45	
2023-05-16 15:03:45,978 skywalking [pid:42] [Thread-4] [DEBUG] reporting segment Segment@140514892420432[segment_id=8c4a7028f3b711eda05b161bd37a2426, spans=[<skywalking.trace.span.EntrySpan object at 0x7fcc2c3f9090>], timestamp=1684220517660, related_traces=[<skywalking.trace.segment._NewID object at 0x7fcc2c3f95d0>]]
2023-05-16 15:03:45	
2023-05-16 15:03:45,978 skywalking [pid:42] [Thread-4] [DEBUG] reporting segment Segment@140514892438992[segment_id=953b4ce8f3b711eda05b161bd37a2426, spans=[<skywalking.trace.span.EntrySpan object at 0x7fcc2c3fd750>], timestamp=1684220532661, related_traces=[<skywalking.trace.segment._NewID object at 0x7fcc2c3fd310>]]
2023-05-16 15:03:45	
2023-05-16 15:03:45,979 skywalking [pid:42] [Thread-4] [DEBUG] reporting segment Segment@140514892485200[segment_id=c1efcb88f3b711eda05b161bd37a2426, spans=[<skywalking.trace.span.EntrySpan object at 0x7fcc2c4091d0>], timestamp=1684220607663, related_traces=[<skywalking.trace.segment._NewID object at 0x7fcc2c4090d0>]]
2023-05-16 15:03:46	
2023-05-16 15:03:46,027 skywalking [pid:1] [MainThread] [DEBUG] Successfully installed plugin sw_django
2023-05-16 15:03:46	
2023-05-16 15:03:46,027 skywalking [pid:1] [MainThread] [DEBUG] installing plugin sw_elasticsearch
```
After calling the interface,the log show nothing difference and segment not be reported.

**Reply by wu-sheng** [reply]
Check the data in the elasticsearch, rather than simply from UI.  Because from my guess, you may face time or timezone issues.

**Reply by Mushano** [reply]
I try to search the segment which segment_id is c1efcb88f3b711eda05b161bd37a2426 that mention above in Elasticsearch,but it is not searchable.

I guess is it possible that there is a blockage when the Python Agent reports to OAP?

PS：Data is available in both versions 0.8 and 1.0 of the Python agent
following is the 0.8 agent, It works welll.
![image](https://github.com/apache/skywalking/assets/38551745/ad1e132c-13e0-43f7-ae3d-a2a33b0a9847)
![image](https://github.com/apache/skywalking/assets/38551745/cb8a7485-7bd3-4a8f-90f0-e2cf9e259d91)

**Reply by Superskyyy** [reply]
> I try to search the segment which segment_id is c1efcb88f3b711eda05b161bd37a2426 that mention above in Elasticsearch,but it is not searchable.
> 
> 
> 
> I guess is it possible that there is a blockage when the Python Agent reports to OAP?
> 
> 
> 
> PS：Data is available in both versions 0.8 and 1.0 of the Python agent
> 
> following is the 0.8 agent, It works welll.
> 
> ![image](https://github.com/apache/skywalking/assets/38551745/ad1e132c-13e0-43f7-ae3d-a2a33b0a9847)
> 
> ![image](https://github.com/apache/skywalking/assets/38551745/cb8a7485-7bd3-4a8f-90f0-e2cf9e259d91)
> 
> 
> 
> 
> 
> 

I will take a look with 1.0.1 and Django later today. Although nothing really changed in 1.0.1 as we just fixed a performance bug. Can you please send the minimal code to reproduce this issue?

**Reply by wu-sheng** [reply]
Agree. 1.0.1 <-> 1.0 change logs, only several bug fixes, https://github.com/apache/skywalking-python/compare/v1.0.0...v1.0.1.

**Reply by Mushano** [reply]
> I will take a look with 1.0.1 and Django later today. Although nothing really changed in 1.0.1 as we just fixed a performance bug. Can you please send the minimal code to reproduce this issue?

I'm really desperate to fix the problem and I'd love to be able to send some code for you to test. But what confuses me is that this problem will **only exist in the test environment**, and I can receive data normally in the local environment. 

![image](https://github.com/apache/skywalking/assets/38551745/06b37d04-7fc9-4649-b972-9cfec4cfa88c)
（1.0.1 in the local environment）

Like the topic I mentioned earlier, https://github.com/apache/skywalking/discussions/10774
this problem also only occurs in test environments (python agent only), even though we control versions of various libraries. 

I will do my best to find a way to reproduce this issue locally and then assist you in fixing it.
